src/core/public/http/http_service.mock.ts

```diff
@@ -36,7 +39,7 @@ const createServiceMock = ({ basePath = '' } = {}): HttpSetupMock => ({
   patch: jest.fn(),
   delete: jest.fn(),
   options: jest.fn(),
-  basePath: new BasePath(basePath),
+  basePath: new BasePath(basePath, basePath, publicBaseUrl),
```

Comment:
Either we need to rename one of the instances of `basePath`, if we need it twice in the args (not sure why though?):
```suggestion
  basePath: new BasePath(basePath, basePathCopy, publicBaseUrl),
```
or if this was just a typo:
```suggestion
  basePath: new BasePath(basePath, publicBaseUrl),
```

Reply:
The constructor of BasePath defaults the second parameter (`serverBasePath`) to be equal to the first which is why I did it this way.

```ts
  constructor(
    private readonly basePath: string = '',
    public readonly serverBasePath: string = basePath,
    public readonly publicBaseUrl?: string
  ) {}
```

I will go ahead and switch this to be `undefined` here so that the behavior is dependent on the implementation and this can't get out of sync.